core/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -0,0 +1,164 @@
+        this.sourceAlias = sourceAlias;
+        this.optionalTargetAlias = optionalTargetAlias;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (sourceAlias == null) {
+            validationException = addValidationError("source alias is missing", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        sourceAlias = in.readString();
+        optionalTargetAlias = in.readOptionalString();
+        int size = in.readVInt();
+        for (int i = 0; i < size; i++) {
+            Condition.ConditionType type = Condition.ConditionType.fromId(in.readByte());
+            long value = in.readLong();
+            this.conditions.add(new Condition(type, value));
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(sourceAlias);
+        out.writeOptionalString(optionalTargetAlias);
+        out.writeVInt(conditions.size());
+        for (Condition condition : conditions) {
+            out.writeByte(condition.getType().getId());
+            out.writeLong(condition.getValue());
+        }
+    }
+
+    @Override
+    public String[] indices() {
+        return new String[] {sourceAlias};
+    }
+
+    @Override
+    public IndicesOptions indicesOptions() {
+        return IndicesOptions.strictSingleIndexNoExpandForbidClosed();
+    }
+
+    public void setSourceAlias(String sourceAlias) {
+        this.sourceAlias = sourceAlias;
+    }
+
+    public void setOptionalTargetAlias(String optionalTargetAlias) {
+        this.optionalTargetAlias = optionalTargetAlias;
+    }
+
+    public void addMaxIndexAgeCondition(String age) {
+        addCondition(Condition.ConditionType.MAX_AGE, Condition.ConditionType.parseFromString(Condition.ConditionType.MAX_AGE, age));
+    }
+
+    public void addMaxIndexDocsCondition(String docs) {
+        addCondition(Condition.ConditionType.MAX_DOCS, Condition.ConditionType.parseFromString(Condition.ConditionType.MAX_DOCS, docs));
+    }
+
+    public void addMaxIndexSizeCondition(String size) {
+        addCondition(Condition.ConditionType.MAX_SIZE, Condition.ConditionType.parseFromString(Condition.ConditionType.MAX_SIZE, size));
+    }
+
+    private void addCondition(Condition.ConditionType conditionType, long value) {
+        this.conditions.add(new Condition(conditionType, value));
+    }
+
+    public List<Condition> getConditions() {
+        return conditions;
+    }
+
+    public String getSourceAlias() {
+        return sourceAlias;
+    }
+
+    public String getOptionalTargetAlias() {
+        return optionalTargetAlias;
+    }
+
+    public void source(BytesReference source) {
+        XContentType xContentType = XContentFactory.xContentType(source);
```

Comment:
I think one thing that ObjectParser would do for this is that it'd make it really simple if we had to add conditions that require more than one value to configure them.